com.unity.shadergraph/Editor/Drawing/Controllers/BlackboardController.cs

```diff
@@ -14,6 +14,47 @@
 
 namespace UnityEditor.ShaderGraph.Drawing
 {
+    struct BlackboardShaderInputOrder
+    {
+        public bool isKeyword;
+        public KeywordType keywordType;
+        public ShaderKeyword builtInKeyword;
+        public string deprecatedPropertyName;
+        public int version;
+    }
+    class BlackboardShaderInputFactory
+    {
+        static public ShaderInput GetShaderInput(BlackboardShaderInputOrder order)
+        {
+            ShaderInput output;
+            if (order.isKeyword)
+            {
+                if (order.builtInKeyword == null)
+                {
+                    output = new ShaderKeyword(order.keywordType);
+                }
+                else
+                {
+                    output = order.builtInKeyword;
+                }
+            }
+            else
+            {
+                switch (order.deprecatedPropertyName)
+                {
+                    case "Color":
+                        output = new ColorShaderProperty(order.version);
+                        break;
+                    default:
+                        //TODO: Not sure what should be the default return property
+                        output = null;
```

Comment:
Thank you @Nightmask3 ! Fixed the typoes haha. I also got another question here, what should we return as default in the switch statement?